### ISSUE TITLE
how to use proxy?

### ISSUE THREAD

**Issue by philos3** [dart-lang/web_socket_channel#86]
how to use proxy?
such as wss://api.huobi.pro/ws  can't connect in china,we must use proxy

**Comment by aswanthtpc** [comment]
got any solution ?

**Comment by shkvoretz** [comment]
we need this also.

**Comment by dudeofawesome** [comment]
There are 2 open PRs to add this: #215 & #211 on `IOWebSocketChannel`s. If you need this now, you can add a dependency pointing to one of those forks:

```yaml
  web_socket_channel:
    git:
      url: git@github.com:iunu/web_socket_channel.git
      ref: feat/custom-client
```

**Comment by shkvoretz** [comment]
@dudeofawesome this is awesome! Many, many thanks!

**Comment by yaohongbin2007** [comment]
> @dudeofawesome this is awesome! Many, many thanks!

@dudeofawesome  The named parameter 'customClient' isn't defined. 
And
[connect](https://api.dart.dev/be/180361/dart-io/WebSocket/connect.html)([String](https://api.dart.dev/be/180361/dart-core/String-class.html) url, {[Iterable](https://api.dart.dev/be/180361/dart-core/Iterable-class.html)<[String](https://api.dart.dev/be/180361/dart-core/String-class.html)>? protocols, [Map](https://api.dart.dev/be/180361/dart-core/Map-class.html)<[String](https://api.dart.dev/be/180361/dart-core/String-class.html), dynamic>? headers, [CompressionOptions](https://api.dart.dev/be/180361/dart-io/CompressionOptions-class.html) compression = CompressionOptions.compressionDefault}) → [Future](https://api.dart.dev/be/180361/dart-async/Future-class.html)<[WebSocket](https://api.dart.dev/be/180361/dart-io/WebSocket-class.html)>
Create a new WebSocket connection. The URL supplied in url must use the scheme ws or wss
do not contain a httpClient param

**Comment by yaohongbin2007** [comment]
> > @dudeofawesome this is awesome! Many, many thanks!
> 
> @dudeofawesome The named parameter 'customClient' isn't defined. And [connect](https://api.dart.dev/be/180361/dart-io/WebSocket/connect.html)([String](https://api.dart.dev/be/180361/dart-core/String-class.html) url, {[Iterable](https://api.dart.dev/be/180361/dart-core/Iterable-class.html)<[String](https://api.dart.dev/be/180361/dart-core/String-class.html)>? protocols, [Map](https://api.dart.dev/be/180361/dart-core/Map-class.html)<[String](https://api.dart.dev/be/180361/dart-core/String-class.html), dynamic>? headers, [CompressionOptions](https://api.dart.dev/be/180361/dart-io/CompressionOptions-class.html) compression = CompressionOptions.compressionDefault}) → [Future](https://api.dart.dev/be/180361/dart-async/Future-class.html)<[WebSocket](https://api.dart.dev/be/180361/dart-io/WebSocket-class.html)> Create a new WebSocket connection. The URL supplied in url must use the scheme ws or wss do not contain a httpClient param

sorry, I find my dart sdk version is low

**Comment by natebosch** [comment]
The `customClient` argument was landed in #259 and is available in version `2.4.0`
